flask_resty/spec/declaration.py

```diff
@@ -0,0 +1,84 @@
+from .utils import get_marshmallow_schema_name, ref
+
+
+class EmptyViewDeclaration(object):
+
+    def __call__(self, view, path, spec):
+        pass
+
+
+class ApiViewDeclaration(EmptyViewDeclaration):
+    """Simple Declaration for ApiView Classes
+    :param many: whether or not this view describes a list or a single instance
+    :param tag: whether or not the schema name should be added as a tag"""
+
+    def __init__(self, many=False, tag=True):
+        self.many = many
+        self.tag = tag
+
+    def __call__(self, view, path, spec):
+        for method in ['get', 'post', 'patch', 'delete', 'put']:
+            if hasattr(view, method):
+                operation = path[method]  # triggers the operation generation
+                if getattr(view, method).__doc__:
+                    # add docstring as description
+                    operation['description'] = getattr(view, method).__doc__
+
+        if not view.schema:
+            return
+
+        schema = get_marshmallow_schema_name(spec, type(view.schema))
+        schema_ref = ref(schema)
+
+        if hasattr(view, 'get'):
+            if not self.many:
+                data = schema_ref
+            else:
+                data = {
+                    'type': 'array',
+                    'items': schema_ref
+                }
+            path['get'].add_property_to_response(prop_name='data', **data)
+
+        if hasattr(view, 'post'):
+            path['post'].add_parameter('body',
+                                       name='body',
+                                       required=True,
+                                       schema=schema_ref)
+            path['post'].declare_response(201)
+
+        if hasattr(view, 'put'):
+            path['put'].add_parameter('body',
+                                      name='body',
+                                      required=True,
+                                      schema=schema_ref)
+            path['put'].declare_response(204)
+
+        if hasattr(view, 'patch'):
+            path['patch'].add_parameter('body',
+                                        name='body',
+                                        required=True,
+                                        schema=schema_ref)
+            path['patch'].add_property_to_response(prop_name='data',
+                                                   **schema_ref)
+
+        if hasattr(view, 'delete'):
+            path['delete'].declare_response(204)
+
+        if self.tag:
+            for method in ['get', 'post', 'patch', 'delete', 'put']:
+                path[method].add_tag(schema)
+
+
+class ModelViewDeclaration(ApiViewDeclaration):
+    """Declaration for Views that specify a model"""
+
+    def __call__(self, view, path, spec):
+        super(ModelViewDeclaration, self)\
+            .__call__(view, path, spec)
```

Comment:
http://stackoverflow.com/questions/12342963/invoking-the-call-method-of-a-superclass